src/gamestate/system_state.cpp

```diff
@@ -754,9 +754,16 @@ void state::render() { // called to render the frame may (and should) delay retu
 	if(game_state_was_updated) {
 		if(!ui_state.tech_queue.empty()) {
 			if(!world.nation_get_current_research(local_player_nation)) {
-				auto tech = ui_state.tech_queue.back();
-				command::start_research(*this, local_player_nation, tech);
-				ui_state.tech_queue.pop_back();
+				auto it = ui_state.tech_queue.begin();
+				if(command::can_start_research(*this, local_player_nation, *it)) {
+					// can research, so research it
+					command::start_research(*this, local_player_nation, *it);
+					ui_state.tech_queue.erase(it);
+				} else {
```

Comment:
don't do this, as it reorders the queue
instead: search for the first tech you can start, from the front to the back
on finding a tech, erase it from the queue, preserving its order
else, do nothing